api.rb

```diff
@@ -1,684 +1,139 @@
+$redis = Redis.new host: ENV.fetch('REDIS_PORT_6379_TCP_ADDR', 'localhost'),
+                   port: ENV.fetch('REDIS_PORT_6379_TCP_PORT', 6379)
 
-  # Set up MySQL DB
-  if host.to_s == ''
-    $client = Mysql2::Client.new(:host => "localhost",
-                                :username => "root",
-                                :database => "fbapp",
-                                :reconnect => true)
-  else
-    # Connect to a MySQL server via a linked docker container
-    $client = Mysql2::Client.new(
-                               :host => "mysql",
-                               :port => ENV['MYSQL_PORT_3306_TCP_PORT'],
-                               :password => ENV['MYSQL_ENV_MYSQL_ROOT_PASSWORD'],
-                               :username => "root",
-                               :database => "fbapp",
-                               :reconnect => true)
-  end
-
-  # Set up Redis for caching
-  $redis = Redis.new(:host => "redis",
-                    :port => ENV['REDIS_PORT_6379_TCP_PORT'],
-                    :timeout => 500,
-                    :connect_timeout => 500)
-
-  # before do
-  #   puts '[Params]'
-  #   p params
-  # end
-
-  # before do
-  #   puts '[env]'
-  #   p env
-  # end
-
-
-  class LogstashLogger < Rack::CommonLogger
-    private
+ActiveRecord::Base.establish_connection($config['db']['fb'])
 
-    ### FIXME UGH, duplicate this definition so it's available to the private class.
-    def redis_exists(key)
-      if !$use_caching
-        return false
-      else
-        begin
-          return $redis.exists(key)
-        rescue
-          return false
-        end
-      end
-    end
-    def redis_set(key, value)
-      if $use_caching
-        begin
-          return $redis.set(key, value)
-        rescue
-          return nil
-        end
-      end
-    end
-    def redis_get(key)
-      begin
-        return $redis.get(key)
-      rescue
-        return nil
-      end
-    end
-
-
-    def ip_anonymize(ip)
-      begin
-        ismatch = ip.match('127.0.0.1|localhost|::1')
-        if ismatch.nil?
-          if redis_exists(ip)
-            out = JSON.parse(redis_get(ip))
-          else
-            res = Geolocater.geolocate_ip(ip)
-            out = res.keep_if { |key, value| key.to_s.match(/city|country_name/) }
-            redis_set(ip, JSON.generate(out))
-          end
-        else
-          out = {"city" => "localhost", "country_name" => "localhost"}
-        end
-        return out
-      rescue
-        return ip
-    #    return {"city" => "localhost", "country_name" => "localhost"}
+class API < Sinatra::Application
+  before do
+    # set headers
+    headers 'Content-Type' => 'application/json; charset=utf8'
+    headers 'Access-Control-Allow-Methods' => 'HEAD, GET'
+    headers 'Access-Control-Allow-Origin' => '*'
+    cache_control :public, :must_revalidate, max_age: 60
+
+    # use redis caching
+    if $config['caching']
+      @cache_key = Digest::MD5.hexdigest(request.url)
```

Comment:
Its actually the full URL, ex: `http://fishbase.ropensci.org/taxa/?limit=10`

Docs here: http://www.sinatrarb.com/intro.html#Accessing%20the%20Request%20Object